banyand/tsdb/segment.go

```diff
@@ -19,42 +19,74 @@ package tsdb
 
 import (
 	"context"
+	"strconv"
 	"sync"
 	"time"
 
 	"github.com/apache/skywalking-banyandb/banyand/kv"
+	"github.com/apache/skywalking-banyandb/banyand/tsdb/bucket"
 	"github.com/apache/skywalking-banyandb/pkg/logger"
 )
 
 type segment struct {
-	path string
+	id     uint16
+	path   string
+	suffix string
 
 	lst         []*block
 	globalIndex kv.Store
 	sync.Mutex
-	l         *logger.Logger
-	startTime time.Time
-	endTime   time.Time
+	l              *logger.Logger
+	reporterStopCh chan struct{}
+	TimeRange
 }
 
-func (s *segment) contains(ts time.Time) bool {
-	greaterAndEqualStart := s.startTime.Equal(ts) || s.startTime.Before(ts)
-	if s.endTime.IsZero() {
-		return greaterAndEqualStart
+func (s *segment) Report() bucket.Channel {
+	ch := make(bucket.Channel)
+	interval := s.Duration() >> 4
```

Comment:
why we need to divide by 2^4?

Reply:
get rid of setting the interval by hand. The interval = the segment duration / 16